sim/core/unit.go

```diff
@@ -773,6 +775,7 @@ func (unit *Unit) reset(sim *Simulation, _ Agent) {
 
 	unit.DynamicStatsPets = unit.DynamicStatsPets[:0]
 	unit.DynamicMeleeSpeedPets = unit.DynamicMeleeSpeedPets[:0]
+	unit.SpellsInFlight = nil
 
```

Comment:
It's more performant to the use the built-in `clear()` function for this:
```
clear(unit.SpellsInFlight)
```

Reply:
clear would delete all the entries but still have the initialized map, even if it's not utilized. I don't know if Unit is a generic struct for every unit in a sim, Player, Targets, "Raiders" etc. So may be better to have it nil since targets and any other "Unit" may not utilize the SpellsInFlight.

Sorry, still not very familiar with the sims workings. Let me know if it's still okay to change to clear.

Reply:
on second thought, SpellsInFlight would only be initialized if it was used, therefore running clear on a nil would be a no-op so it wouldnt do anything. making that change now.